### DISCUSSION TITLE
JavaScript 모듈 번들러에 대해 설명해 주세요.

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#107] [모듈]
- JavaScript 모듈 번들러에 대해 설명해 주세요.
  - Webpack의 주요 기능을 설명해 주세요.

**Comment by publdaze** [comment]
> * JavaScript 모듈 번들러에 대해 설명해 주세요.
- 모듈 번들러는 여러개의 파일을 하나의 파일로 병합하여 로드할 수 있게 도와줌
>   * Webpack의 주요 기능을 설명해 주세요.
- 모듈 번들링 - 여러개의 파일을 하나의 파일로 병합하여, HTTP 요청의 수를 줄일 수 있음
- 코드분할, 레이지 로딩, 트리 쉐이킹 등의 기능을 제공해 로딩 시간을 단축시키고 불필요한 코드 실행을 방지하는 등 성능 향상 가능

**Comment by pipisebastian** [comment]
> JavaScript 모듈 번들러에 대해 설명해 주세요.


- 모듈 번들러는, 모듈의 의존성을 분석하여, 파일로 병합해주는 도구입니다.
- 이는, 개발자는 모듈 단위로 편리하게 개발이 가능하며, 모듈러는 하나의 js 파일로 번들링해주기에, 1번의 네트워크 요청으로 웹 페이지를 로드할 수 있습니다.

> * Webpack의 주요 기능을 설명해 주세요.

- 많이 사용되는 js 모듈 번들러 중 하나입니다. 앞서 말한 모듈 번들러의 특징을 가지고 있습니다.
- js 와 함께, css, image 등의 리소스의 의존성도 함께 관리
- Tree shaking 과 같은 최적화 수행
  - 코드 축소 + 사용하지 않는 코드 제거
- Code splitting 
  - 필요한 시점에서, 리소스를 다운받아 실행 -> 속도개선
